praw/models/reddit/subreddit.py

```diff
@@ -2155,3 +2173,46 @@ def revisions(self, **generator_kwargs):
         url = API_PATH['wiki_revisions'].format(subreddit=self.subreddit)
         return WikiPage._revision_generator(
             self.subreddit, url, generator_kwargs)
+
+
+class SubredditEmoji(object):
+    """Provides a set of functions to a Subreddit for emoji."""
+
+    def __getitem__(self, name):
+        """Lazily return the Emoji for the subreddit named ``name``.
+
+        This method is to be used to fetch a specific emoji url, like so:
+
+        .. code:: python
+
+           emoji = reddit.subreddit('iama').emoji['cake']
+           print(emoji)
+
+        """
+        return Emoji(self.subreddit._reddit, self.subreddit, name.lower())
+
+    def __init__(self, subreddit):
+        """Create a SubredditModeration instance.
+
+        :param subreddit: The subreddit to moderate.
+
+        """
+        self.subreddit = subreddit
+
+    def __iter__(self):
+        """Iterate through the Emoji for the subreddit.
+
+        This method is to be used to discover all emoji for a subreddit:
+
+        .. code:: python
+
+           for emoji in reddit.subreddit('iama').emoji:
+               print(emoji)
+
+        """
+        response = self.subreddit.fullname._reddit.get(
```

Comment:
`response = self.subreddit.fullname._reddit.get(` should be `response = self.subreddit._reddit.get(` — `self.subreddit.fullname` doesn't have a `_reddit` attribute. It seems like you just got caught up in an overzealous find-and-replace.